pkg/ingress/utils/utils.go

```diff
@@ -74,14 +74,17 @@ func stringSlicesEqual(x, y []string) bool {
 
 // GetNodeID get instance ID from the Node spec.
 func GetNodeID(node *apiv1.Node) (string, error) {
-	var providerIDRegexp = regexp.MustCompile(`^openstack:///([^/]+)$`)
+	var providerIDRegexp = regexp.MustCompile(`^openstack://([^/]*)/([^/]+)$`)
 
 	matches := providerIDRegexp.FindStringSubmatch(node.Spec.ProviderID)
-	if len(matches) != 2 {
+	switch len(matches) {
```

Comment:
This is a bit weird. Just use a non-capturing regex group (`(?: ... )`) above instead, and then always use the original len == 2 code here.

`^openstack://(?:[^/]*)/([^/]+)$`